### ISSUE TITLE
The visualization results after training have not changed

### ISSUE THREAD

**Issue by chaoshuoZhang** [rohitgandikota/erasing#36]
Hello，the parameters obtained by training the new code **esd_diffuser.py**. Why are the images generated by using **generate_images.py** the same as the results of SDv1-4? The prompt I used during training was 'van Gogh', and the prompt during generation was **vangogh_prompts.csv**

**Comment by rohitgandikota** [comment]
aah we did not update the generate.py file. Good catch! 

For now - please look at the inference.ipynb to see how to generate with the current code version

**Comment by chaoshuoZhang** [comment]
Thank you, I found that if I use the model parameters saved with the new version of the code directly, if I use **generate_images.py line 66 unet.load_state_dict(torch.load(model_path))**, it is unable to update the parameters after fine-tuning. However, this error does not affect the operation of the code. I wrote a new py file with reference to generate_images.py and inference.ipynb.I share this here hoping it will inspire others.

```from transformers import CLIPTextModel, CLIPTokenizer
from diffusers import AutoencoderKL, UNet2DConditionModel, PNDMScheduler
from diffusers import LMSDiscreteScheduler
import torch
from PIL import Image
import pandas as pd
import argparse
import os
from utils.utils import *
import torch


def generate_images(model_name, prompts_path, save_path, device='cuda:0', guidance_scale=7.5, image_size=512,
                    ddim_steps=100, num_samples=10, from_case=0):
    df = pd.read_csv(prompts_path)

    folder_path = f'{save_path}/{model_name}'
    os.makedirs(folder_path, exist_ok=True)

    for _, row in df.iterrows():
        prompt = [str(row.prompt)] * num_samples
        seed = row.evaluation_seed
        case_number = row.case_number
        if case_number < from_case:
            continue

        num_inference_steps = ddim_steps  # Number of denoising steps

        guidance_scale = guidance_scale  # Scale for classifier-free guidance

        generator = torch.manual_seed(seed)  # Seed generator to create the inital latent noise

        batch_size = len(prompt)
        with finetuner:
            pil_images = diffuser(prompt,
                                  img_size=image_size,
                                  n_steps=num_inference_steps,
                                  n_imgs=1,
                                  generator=generator,
                                  guidance_scale=guidance_scale
                                  )[0]

        # image = (image / 2 + 0.5).clamp(0, 1)
        # image = image.detach().cpu().permute(0, 2, 3, 1).numpy()
        # images = (image * 255).round().astype("uint8")
        # pil_images = [Image.fromarray(image) for image in images]
        for num, im in enumerate(pil_images):
            im.save(f"{folder_path}/{case_number}_{num}.png")


if __name__ == '__main__':
    parser = argparse.ArgumentParser(
        prog='generateImages',
        description='Generate Images using Diffusers Code')
    parser.add_argument('--model_name', help='name of model', type=str, required=True)
    parser.add_argument('--train_method', help='name of mode', type=str, required=True)
    parser.add_argument('--prompts_path', help='path to csv file with prompts', type=str, required=True)
    parser.add_argument('--save_path', help='folder where to save images', type=str, required=True)
    parser.add_argument('--device', help='cuda device to run on', type=str, required=False, default='cuda:0')
    parser.add_argument('--guidance_scale', help='guidance to run eval', type=float, required=False, default=7.5)
    parser.add_argument('--image_size', help='image size used to train', type=int, required=False, default=512)
    parser.add_argument('--from_case', help='continue generating from case_number', type=int, required=False, default=0)
    parser.add_argument('--num_samples', help='number of samples per prompt', type=int, required=False, default=1)
    parser.add_argument('--ddim_steps', help='ddim steps of inference used to train', type=int, required=False,
                        default=100)
    args = parser.parse_args()

    model_name = args.model_name
    prompts_path = args.prompts_path
    save_path = args.save_path
    device = args.device
    guidance_scale = args.guidance_scale
    image_size = args.image_size
    ddim_steps = args.ddim_steps
    num_samples = args.num_samples
    from_case = args.from_case
    train_method = args.train_method
    diffuser = StableDiffuser(scheduler='DDIM').to(device)

    finetuner = FineTunedModel(diffuser, train_method=train_method)
    esd_path = f'models/{model_name}.pt'
    finetuner.load_state_dict(torch.load(esd_path))

    generate_images(model_name, prompts_path, save_path, device=device,
                    guidance_scale=guidance_scale, image_size=image_size, ddim_steps=ddim_steps,
                    num_samples=num_samples, from_case=from_case)

    # python generate-images.py --model_name esd-vangogh_from_vangogh-xattn_1-epochs_200 --prompts_path data/vangogh_prompts.csv --save_path images/--device cuda:2
